src/main/java/com/ksabbak/javaserver/storage/Store.java

```diff
@@ -0,0 +1,47 @@
+package com.ksabbak.javaserver.storage;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Store implements Persistable {
+    private Map<String, String> store;
+
+    public Store(){
+        store = new HashMap<String, String>();
+    }
+
+    @Override
+    synchronized public String create(String key) throws Exception {
+        return create(key, "");
+    }
+
+    @Override
+    synchronized public String create(String key, String value) throws Exception {
+        if (store.get(key) != null) throw new Exception();
+        store.put(key, value);
+        return key;
+    }
+
+    @Override
+    synchronized public String update(String key, String value) throws Exception{
+        if (store.get(key) == null) throw new Exception();
+        store.put(key, value);
+        return key;
+    }
+
+    @Override
+    public String read(String key) throws Exception{
+        String value = store.get(key);
+        if (value == null) throw new Exception();
+        return value;
+    }
+
+    @Override
+    synchronized public String delete(String key) throws Exception{
+        String value = store.remove(key);
+        if (value == null){
+            throw new Exception();
+        }
+        return value;
```

Comment:
It's a bit odd to have delete return the deleted value. I would personally have the consumer call `read` then `delete` if they both needed the value and wanted to delete it.

Reply:
Part of this was me being really dedicated to the idea of these methods being pretty parallel and the other was that it mimics the remove method. Those aren't really good reasons though so I'll get rid of the return string.